### PR TITLE
feat: Add arguments to gen.py post-deploy

### PR DESCRIPTION
**Pull request by rayjh**

### CODE REVIEW

File: scripts/python/gen.py [hunk 1]
```diff
@@ -485,6 +488,7 @@ def _run_playbook(playbook):
 
 if __name__ == '__main__':
     args = argparse_gen.get_parsed_args()
+    print(args)
```
Comment (jaywcarman) [inline scripts/python/gen.py]: Why print the args here?

File: scripts/gen [hunk 2]
```diff
@@ -74,17 +74,17 @@ if [[ "$1" == "post-deploy" ]] && ([[ "$2" == "-a" ]] || [[ "$2" == "--all" ]] |
...
+    gen.py post-deploy --ssh-keyscan $@
     if [ $? -ne 0 ]; then
         exit
     fi
 
-    gen.py deploy --gather-mac-addr $@
+    gen.py post-deploy --gather-mac-addr $@
     if [ $? -ne 0 ]; then
         exit
     fi
```
Comment (jaywcarman) [inline scripts/gen]: Also need to include:
gen.py post-deploy --lookup-interface-names $@

I forgot to add this to 'gen' in my commit.

File: scripts/gen-ac [hunk 3]
```diff
@@ -55,6 +55,14 @@ _gen()
         return 0
     fi
 
+    opts="-h --help --ssh-keyscan --gather-mac-addr --config-client-os "
```
Comment (jaywcarman) [inline scripts/gen-ac]: --lookup-interface-names

File: scripts/python/lib/argparse_gen.py [hunk 4]
```diff
@@ -271,13 +264,39 @@ def get_args(parser_args=False):
...
+        metavar='CONTAINER-NAME',
+        help='Scan SSH keys')
+
+    parser_post_deploy.add_argument(
+        '--gather-mac-addr',
+        nargs='?',
+        default=ABSENT,
+        metavar='CONTAINER-NAME',
+        help='Gather MAC addresses from switches and update inventory')
+
```
Comment (jaywcarman) [inline scripts/python/lib/argparse_gen.py]: --lookup-interface-names

File: scripts/python/lib/argparse_gen.py [hunk 5]
```diff
@@ -313,14 +332,22 @@ def _check_deploy(args, subparser):
...
             args.gather_mac_addr == ABSENT and
-            args.lookup_interface_names == ABSENT and
             args.config_client_os == ABSENT and
             args.all == ABSENT):
         subparser.error(
             'one of the arguments --create-inventory --install-cobbler'
             ' --inv-add-ports-pxe --inv-add-ports-ipmi --download-os-images'
             ' --add-cobbler-distros --add-cobbler-systems --install-client-os'
-            ' --ssh-keyscan --gather-mac-addr --lookup-interface-names '
+            ' --ssh-keyscan --gather-mac-addr --config-client-os -a/--all is'
```
Comment (jaywcarman) [inline scripts/python/lib/argparse_gen.py]: ' --ssh-keyscan --gather-mac-addr --config-client-os' should not be included in '_check_deploy' if they're moving to '_check_post_deploy'.

File: scripts/python/lib/argparse_gen.py [hunk 6]
```diff
@@ -313,14 +332,22 @@ def _check_deploy(args, subparser):
...
+            ' required')
+
+
+def _check_post_deploy(args, subparser):
+    if (args.ssh_keyscan == ABSENT and args.gather_mac_addr == ABSENT and
+            args.config_client_os == ABSENT and args.data_switches == ABSENT and
+            args.all == ABSENT):
+        subparser.error(
+            'one of the arguments --ssh-keyscan --gather-mac-addr'
             '--config-client-os -a/--all is required')
```
Comment (jaywcarman) [inline scripts/python/lib/argparse_gen.py]: --data-swit

File: scripts/python/gen.py (hunk 1 above)
Comment (rayjh) [reply scripts/python/gen.py]: Remnant.  I'll remove it.

File: scripts/gen (hunk 2 above)
Comment (rayjh) [reply scripts/gen]: np.  I'll ad it

**Pull request merged**